From b6da8a453f81f3f233a94b3dd88420a18ab4adff Mon Sep 17 00:00:00 2001
From: costi <constantin.tudor@freescale.com>
Date: Tue, 30 May 2017 15:09:24 +0300
Subject: [PATCH 224/767] fsl-backplane: Get backplane link status from pcs

-revert define value change

Signed-off-by: costi <constantin.tudor@freescale.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/phy/fsl_backplane.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/net/phy/fsl_backplane.c b/drivers/net/phy/fsl_backplane.c
index ba73d4219a78..76865261c15d 100644
--- a/drivers/net/phy/fsl_backplane.c
+++ b/drivers/net/phy/fsl_backplane.c
@@ -40,6 +40,9 @@
 #define TRAIN_DISABLE				0x1
 #define RX_STAT					0x1
 
+#define FSL_KR_RX_LINK_STAT_MASK		0x1000
+#define FSL_XFI_PCS_10GR_SR1                    0x20
+
 /* Freescale KX PCS mode register */
 #define FSL_PCS_IF_MODE				0x8014
 
@@ -784,10 +787,10 @@ static int is_link_up(struct phy_device *phydev)
 {
 	int val;
 
-	phy_read_mmd(phydev, MDIO_MMD_AN, MDIO_STAT1);
-	val = phy_read_mmd(phydev, MDIO_MMD_AN, MDIO_STAT1);
+	phy_read_mmd(phydev, MDIO_MMD_PCS, FSL_XFI_PCS_10GR_SR1);
+	val = phy_read_mmd(phydev, MDIO_MMD_PCS, FSL_XFI_PCS_10GR_SR1);
 
-	return (val & AN_LNK_UP_MASK) ? 1 : 0;
+	return (val & FSL_KR_RX_LINK_STAT_MASK) ? 1 : 0;
 }
 
 static int is_link_training_fail(struct phy_device *phydev)
-- 
2.17.0

